### PR TITLE
fix(curriculum): Add solutions to 16 Applied Accessibility challenges #35751

### PR DESCRIPTION
**Pull request by effendiian**
This commit adds 16 full working solutions to the responsive web design
principles dealing with applied accessibility.

- [x] I have read [freeCodeCamp's contribution guidelines](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/CONTRIBUTING.md).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `master` branch of freeCodeCamp.
- [x] None of my changes are plagiarized from another source without proper attribution.
- [x] All the files I changed are in the same world language (for example: only English changes, or only Chinese changes, etc.)
- [x] My changes do not use shortened URLs or affiliate links.

Related to issue #35751

### CODE REVIEW

File: curriculum/challenges/english/01-responsive-web-design/applied-accessibility/give-links-meaning-by-using-descriptive-link-text.english.md [hunk 1]
```diff
@@ -57,6 +57,14 @@ tests:
 <section id='solution'>
 
 ```js
```
Comment (thecodingaviator) [inline curriculum/challenges/english/01-responsive-web-design/applied-accessibility/give-links-meaning-by-using-descriptive-link-text.english.md]:
```suggestion
```html
```

File: curriculum/challenges/english/01-responsive-web-design/applied-accessibility/improve-accessibility-of-audio-content-with-the-audio-element.english.md [hunk 2]
```diff
@@ -73,6 +73,16 @@ tests:
 <section id='solution'>
 
 ```js
```
Comment (thecodingaviator) [inline curriculum/challenges/english/01-responsive-web-design/applied-accessibility/improve-accessibility-of-audio-content-with-the-audio-element.english.md]:
```suggestion
```html
```

File: curriculum/challenges/english/01-responsive-web-design/applied-accessibility/improve-chart-accessibility-with-the-figure-element.english.md [hunk 3]
```diff
@@ -99,6 +99,46 @@ tests:
 <section id='solution'>
 
 ```js
```
Comment (thecodingaviator) [inline curriculum/challenges/english/01-responsive-web-design/applied-accessibility/improve-chart-accessibility-with-the-figure-element.english.md]:
```suggestion
```html
```

File: curriculum/challenges/english/01-responsive-web-design/applied-accessibility/improve-form-field-accessibility-with-the-label-element.english.md [hunk 4]
```diff
@@ -83,6 +83,37 @@ tests:
 <section id='solution'>
 
 ```js
```
Comment (thecodingaviator) [inline curriculum/challenges/english/01-responsive-web-design/applied-accessibility/improve-form-field-accessibility-with-the-label-element.english.md]:
```suggestion
```html
```

File: curriculum/challenges/english/01-responsive-web-design/applied-accessibility/improve-readability-with-high-contrast-text.english.md [hunk 5]
```diff
@@ -66,6 +66,23 @@ tests:
 <section id='solution'>
 
 ```js
```
Comment (thecodingaviator) [inline curriculum/challenges/english/01-responsive-web-design/applied-accessibility/improve-readability-with-high-contrast-text.english.md]:
```suggestion
```html
```

File: curriculum/challenges/english/01-responsive-web-design/applied-accessibility/jump-straight-to-the-content-using-the-main-element.english.md [hunk 6]
```diff
@@ -57,6 +57,12 @@ tests:
 <section id='solution'>
 
 ```js
```
Comment (thecodingaviator) [inline curriculum/challenges/english/01-responsive-web-design/applied-accessibility/jump-straight-to-the-content-using-the-main-element.english.md]:
```suggestion
```html
```

File: curriculum/challenges/english/01-responsive-web-design/applied-accessibility/know-when-alt-text-should-be-left-blank.english.md [hunk 7]
```diff
@@ -63,6 +63,17 @@ tests:
 <section id='solution'>
 
 ```js
```
Comment (thecodingaviator) [inline curriculum/challenges/english/01-responsive-web-design/applied-accessibility/know-when-alt-text-should-be-left-blank.english.md]:
```suggestion
```html
```

File: curriculum/challenges/english/01-responsive-web-design/applied-accessibility/make-elements-only-visible-to-a-screen-reader-by-using-custom-css.english.md [hunk 8]
```diff
@@ -143,6 +143,89 @@ tests:
 <section id='solution'>
 
 ```js
```
Comment (thecodingaviator) [inline curriculum/challenges/english/01-responsive-web-design/applied-accessibility/make-elements-only-visible-to-a-screen-reader-by-using-custom-css.english.md]:
```suggestion
```html
```

File: curriculum/challenges/english/01-responsive-web-design/applied-accessibility/make-links-navigable-with-html-access-keys.english.md [hunk 9]
```diff
@@ -76,6 +76,27 @@ tests:
 <section id='solution'>
 
 ```js
```
Comment (thecodingaviator) [inline curriculum/challenges/english/01-responsive-web-design/applied-accessibility/make-links-navigable-with-html-access-keys.english.md]:
```suggestion
```html
```

File: curriculum/challenges/english/01-responsive-web-design/applied-accessibility/make-screen-reader-navigation-easier-with-the-footer-landmark.english.md [hunk 10]
```diff
@@ -83,6 +83,39 @@ tests:
 <section id='solution'>
 
 ```js
```
Comment (thecodingaviator) [inline curriculum/challenges/english/01-responsive-web-design/applied-accessibility/make-screen-reader-navigation-easier-with-the-footer-landmark.english.md]:
```suggestion
```html
```

File: curriculum/challenges/english/01-responsive-web-design/applied-accessibility/make-screen-reader-navigation-easier-with-the-header-landmark.english.md [hunk 11]
```diff
@@ -78,6 +78,30 @@ tests:
 <section id='solution'>
 
 ```js
```
Comment (thecodingaviator) [inline curriculum/challenges/english/01-responsive-web-design/applied-accessibility/make-screen-reader-navigation-easier-with-the-header-landmark.english.md]:
```suggestion
```html
```

File: curriculum/challenges/english/01-responsive-web-design/applied-accessibility/make-screen-reader-navigation-easier-with-the-nav-landmark.english.md [hunk 12]
```diff
@@ -83,6 +83,36 @@ tests:
 <section id='solution'>
 
 ```js
```
Comment (thecodingaviator) [inline curriculum/challenges/english/01-responsive-web-design/applied-accessibility/make-screen-reader-navigation-easier-with-the-nav-landmark.english.md]:
```suggestion
```html
```

File: curriculum/challenges/english/01-responsive-web-design/applied-accessibility/use-tabindex-to-add-keyboard-focus-to-an-element.english.md [hunk 13]
```diff
@@ -97,6 +97,52 @@ tests:
 <section id='solution'>
 
 ```js
```
Comment (thecodingaviator) [inline curriculum/challenges/english/01-responsive-web-design/applied-accessibility/use-tabindex-to-add-keyboard-focus-to-an-element.english.md]:
```suggestion
```html
```

File: curriculum/challenges/english/01-responsive-web-design/applied-accessibility/use-tabindex-to-specify-the-order-of-keyboard-focus-for-several-elements.english.md [hunk 14]
```diff
@@ -87,6 +87,36 @@ tests:
 <section id='solution'>
 
 ```js
```
Comment (thecodingaviator) [inline curriculum/challenges/english/01-responsive-web-design/applied-accessibility/use-tabindex-to-specify-the-order-of-keyboard-focus-for-several-elements.english.md]:
```suggestion
```html
```

File: curriculum/challenges/english/01-responsive-web-design/applied-accessibility/wrap-content-in-the-article-element.english.md [hunk 15]
```diff
@@ -72,6 +72,26 @@ tests:
 <section id='solution'>
 
 ```js
```
Comment (thecodingaviator) [inline curriculum/challenges/english/01-responsive-web-design/applied-accessibility/wrap-content-in-the-article-element.english.md]:
```suggestion
```html
```

File: curriculum/challenges/english/01-responsive-web-design/applied-accessibility/wrap-radio-buttons-in-a-fieldset-element-for-better-accessibility.english.md [hunk 16]
```diff
@@ -100,6 +100,48 @@ tests:
 <section id='solution'>
 
 ```js
```
Comment (thecodingaviator) [inline curriculum/challenges/english/01-responsive-web-design/applied-accessibility/wrap-radio-buttons-in-a-fieldset-element-for-better-accessibility.english.md]:
```suggestion
```html
```

**Review by thecodingaviator — DISMISSED** [review]
The HTML challenges should have \`\`\`html instead of ```js

**Comment by rimij405** [comment]
Hi @thecodingaviator - the changes you suggested were applied. Let me know if further work needs to be done on these. Thank you.

**Comment by RandellDawson** [comment]
@rimij405 Can you squash the commits in a rebase?

**Comment by rimij405** [comment]
Yes. Would I just push after the rebase or do I need to make a new pull request for that?

**Comment by RandellDawson** [comment]
You would do this on this PR.  You can squash the 17 commits down to a single commit.  It cleans up the PR.

**Comment by rimij405** [comment]
Ah, gotcha. I'll get right to that.

**Comment by rimij405** [comment]
@RandellDawson - History for this pull request has been rebased, commits squashed, and changes pushed to the branch. Pending the Travis CI check, everything should be in order now. The `git push -f` clarification helped because I wasn't sure if I should have merged the existing remote (since it couldn't fast-forward) or if I should overwrite it.

Thank you for the guidance!

**Comment by Tanmay-jana** [comment]
Hi

**Comment by Tanmay-jana** [comment]
Hi

**Comment by Tanmay-jana** [comment]
Hi

**Review by ojeytonwilliams — APPROVED** [review]
LGTM :+1:

**Comment by ojeytonwilliams** [comment]
@thecodingaviator If you're happy with the changes, I think this is ready to merge.

**Review by thecodingaviator — APPROVED** [review]
LGTM

**Comment by thecodingaviator** [comment]
Thank you for your contribution to the page! 👍
We're happy to accept these changes, and look forward to future contributions. 📝

**Pull request merged**